crates/topos-tce/src/app_context.rs

```diff
@@ -702,36 +756,76 @@ impl AppContext {
-                                        &self.network_client.local_peer_id, &from_peer, &certificate_id
-                                    );
-                                    self.tce_cli.get_double_echo_channel()
-                                });
+                                        sender = from.to_string(),
+                                            "peer_id {} on_net_event TceCommands::OnReady from peer {} cert id: {}",
+                                            &self.network_client.local_peer_id, &from_peer, &certificate_id
+                                        );
+                                        self.tce_cli.get_double_echo_channel()
+                                    });
 
-                                spawn(self.network_client.respond_to_request(
-                                    NetworkMessage::from(TceCommands::OnDoubleEchoOk {
-                                        from_peer: my_peer,
-                                    }),
-                                    channel,
-                                ));
-                                command_sender
-                                    .send(DoubleEchoCommand::Ready {
-                                        from_peer,
-                                        certificate_id,
-                                        ctx: span.context(),
-                                    })
-                                    .with_context(context)
-                                    .instrument(span)
+                                    spawn(self.network_client.respond_to_request(
+                                        NetworkMessage::from(TceCommands::OnDoubleEchoOk {
+                                            from_peer: my_peer,
+                                        }),
+                                        channel,
+                                    ));
+                                    command_sender
+                                        .send(DoubleEchoCommand::Ready {
+                                            from_peer,
+                                            certificate_id,
+                                            ctx: span.context(),
+                                        })
+                                        .with_context(context)
+                                        .instrument(span)
+                                        .await
+                                        .expect("Receive the Ready");
+                                } else if self
+                                    .pending_storage
+                                    .get_certificate(certificate_id)
                                     .await
-                                    .expect("Receive the Ready");
+                                    .is_ok()
+                                {
+                                    info!("Received Ready for already delivered certificate {certificate_id}, ignoring");
```

Comment:
actually this case is not an issue and shouldn't be ignored since we need to process the Ready even if we already delivered, until we send our own Ready

Reply:
`.editorconfig` is a generic file to be used by multiple editors not just one.

Reply:
that's why It is in `info`. You think I should remove it?

Reply:
```suggestion
                                    info!("Received Ready for already delivered certificate {certificate_id}");
```

I'd remove the "ignoring" then